django_codemod/visitors/models.py

```diff
@@ -129,3 +131,48 @@ def leave_Return(
                 value=Call(func=Name("reverse"), args=args)
             )
         return super().leave_Return(original_node, updated_node)
+
+
+def is_foreign_key(node: Call) -> bool:
+    return m.matches(node, m.Call(func=m.Attribute(attr=m.Name(value="ForeignKey"))))
+
+
+def is_one_to_one_field(node: Call) -> bool:
+    return m.matches(
+        node, m.Call(func=m.Attribute(attr=m.Name(value="OneToOneField")),),
+    )
+
+
+def has_on_delete(node: Call) -> bool:
+    # if on_delete exists in any kwarg we return True
+    for arg in node.args:
+        if m.matches(arg.keyword, m.Name("on_delete")):
```

Comment:
I tried to do this with an argument matcher in lots of different ways, but couldn't get it to work:
```
> node.args[1]
Arg(
    value=Attribute(
        value=Name(
            value='models',
            lpar=[],
            rpar=[],
        ),
        attr=Name(
            value='CASCADE',
            lpar=[],
            rpar=[],
        ),
        dot=Dot(
            whitespace_before=SimpleWhitespace(
                value='',
            ),
            whitespace_after=SimpleWhitespace(
                value='',
            ),
        ),
        lpar=[],
        rpar=[],
    ),
    keyword=Name(
        value='on_delete',
        lpar=[],
        rpar=[],
    ),
    equal=AssignEqual(
        whitespace_before=SimpleWhitespace(
            value=' ',
        ),
        whitespace_after=SimpleWhitespace(
            value=' ',
        ),
    ),
    comma=MaybeSentinel.DEFAULT,
    star='',
    whitespace_after_star=SimpleWhitespace(
        value='',
    ),
    whitespace_after_arg=SimpleWhitespace(
        value='',
    ),
)

> node.args[1].keyword
Name(
    value='on_delete',
    lpar=[],
    rpar=[],
)
> m.matches(node.args[1].keyword, m.Name("on_delete"))                                                   
True
> m.matches(node, m.Call(args=(m.OneOf(m.Arg(keyword=m.Name(value="on_delete"))))))
False
> m.matches(node, m.Call(args=(m.Arg(keyword=m.Name(value="on_delete")))))
False
```

I could never get it to detect properly until I switched to this pattern 🤔

Reply:
Hum, I'll take a quick look in a bit.

Reply:
Ok, so looks like we could match the `arg`:

```
for arg in node.args:
	if m.matches(arg, m.Arg(keyword=m.Name("on_delete"))):
        return True
```

But yes, I think it's a sort of limitation of libCST. I didn't find an way to match an element of a sequence, and it looks like we sometimes need to do the matching in a for loop.

It could be a concern for speed, but in this case it's probably fine. It will only be evaluated only for `ForeignKey` or `OneToOneField`, since it's in the second part of the `and` at line 166.

Reply:
thanks for investigating! Glad i'm not crazy 😆